From b7aecfe5f911920a42b42677ef82bcb588096c0d Mon Sep 17 00:00:00 2001
From: Johannes Berg <johannes.berg@intel.com>
Date: Wed, 8 May 2024 22:15:43 +0200
Subject: [PATCH] wlantest: ICMP: Use sta_find_mlo()

Using just sta_find() won't work for any link addresses
but the assoc link, use sta_find_mlo() instead.

Signed-off-by: Johannes Berg <johannes.berg@intel.com>
---
 wlantest/rx_ip.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/wlantest/rx_ip.c b/wlantest/rx_ip.c
index b0fdd2068..5dc186c45 100644
--- a/wlantest/rx_ip.c
+++ b/wlantest/rx_ip.c
@@ -85,7 +85,7 @@ static void rx_data_icmp(struct wlantest *wt, const u8 *bssid,
 	if (sta_addr == NULL)
 		return; /* FromDS broadcast ping */
 
-	sta = sta_find(bss, sta_addr);
+	sta = sta_find_mlo(wt, bss, sta_addr);
 	if (sta == NULL) {
 		add_note(wt, MSG_INFO, "No STA " MACSTR
 			 " known for ICMP packet", MAC2STR(sta_addr));
@@ -93,7 +93,7 @@ static void rx_data_icmp(struct wlantest *wt, const u8 *bssid,
 	}
 
 	ping_update(wt, sta, hdr->type == ICMP_ECHO, src, dst, id, seq);
-	if (peer_addr && (sta = sta_find(bss, peer_addr)))
+	if (peer_addr && (sta = sta_find_mlo(wt, bss, peer_addr)))
 		ping_update(wt, sta, hdr->type == ICMP_ECHO, src, dst, id, seq);
 }
 
-- 
2.43.0

